sycl/doc/PreprocessorMacros.md

```diff
@@ -1,6 +1,18 @@
 # Preprocessor Macros 
 
-This file describes macros that have effect on SYCL compiler and run-time.
+## Predefined macros
+
+This section describes macros that are predefined in SYCL library.
+
+### __SYCL_COMPILER_VERSION
+
+This macro contains the build date of SYCL library and is used as a versioning
```

Comment:
> *This macro contains the build date of SYCL library*

What is the rationale?
First this is about the library and not the compiler...
Then this is the build date... Who cares? For example if I compile in 2025 a work-tree of today, I will have SYCL 2025?
Can I change the time of my computer to get a preview of SYCL 2030? :-)

Reply:
I am agree that this is imperfect way of versioning SYCL library. If improvement of this versioning is required then we should create a request.
However, some users are going to use this macro for splitting code depending on SYCL version now because there is no other way to check it yet. Current PR is for documenting available option to date.

Reply:
What about using the commit date of the library source or equivalent instead of the build date?

Reply:
I suppose that this env variable might be then renamed and updated via separate commits once a day (maybe automatically) or less frequently.
Anyway, the mechanism of such versioning is not clear for me now and this is out of the scope of current PR because it is mentioned as doc update only.
Thank you very much for your suggestion, I've created an issue https://github.com/intel/llvm/issues/2250 to address your concerns regarding current implementation of this variable.

Reply:
We can't use this name.  We have to use something intel/llvm specific, this name implies this is a SYCL official name.  

Also, the 'build date' as mentioned is foolish, we should at least be picking up the clang-version # via CMake, like we do with a bunch of other stuff.  More likely, we probably want our own.